### PR TITLE
IOT: update_thing() now behaves correctly with merge=True

### PR DESCRIPTION
**Pull request by Wurstnase**
The last patch missed some cases and the behaviour was wrong.
I added all those cases in the test and fixed the problems.

### CODE REVIEW

File: moto/iot/models.py [hunk 1]
```diff
@@ -883,7 +883,11 @@ def update_thing(
             if not do_merge:
                 thing.attributes = attributes
             else:
-                thing.attributes = {k: v for k, v in attributes.items() if v}
+                for k, v in attributes.items():
+                    if v:
+                        thing.attributes[k] = v
+                    else:
+                        thing.attributes.pop(k)
```
Comment (bblommers) [inline moto/iot/models.py:890]: How does AWS behave if `k` does not exist here?
I.e., thing.attributes only has attribute`k1`, and a user calls `update_thing(.. {"k2": ""})`?

Comment (Wurstnase) [reply moto/iot/models.py:890]: No exception, nothing changed. Simply a `{'ResponseMetadata': {..., 'HTTPStatusCode': 200,...}`.

Comment (Wurstnase) [reply moto/iot/models.py:890]: I just check another special case.

With `"merge": False` and `update_thing(.. {"k2": ""})`, all attributes are deleted. So an empty value will never be saved.

**Review by bblommers — APPROVED** [review]
Thanks for the update @Wurstnase!

**Pull request merged**